controllers/rest/controller_standard.go

```diff
@@ -681,3 +684,124 @@ func HealthCheck(api *RESTAPI, projection projections.Projection, commandDispatc
+				if !strings.Contains(openIdConnectUrl, ".well-known/openid-configuration") {
+					api.EchoInstance().Logger.Warnf("invalid open id connect url: %s", openIdConnectUrl)
+				} else {
+					//get the Jwk url from open id connect url
+					jwksUrl, err := GetJwkUrl(openIdConnectUrl)
+					if err != nil {
+						api.EchoInstance().Logger.Errorf("unexpected error getting the jwks url: %s", err)
+					} else {
+						//by default skipExpiryCheck is false meaning it will not run an expiry check
+						skipExpiryCheck := false
+						//get skipexpirycheck that is an extension in the openapi spec
+						if expireCheck, ok := schemes.Value.ExtensionProps.Extensions[SKIPEXPIRYCHECKEXTENSION]; ok {
+							err := json.Unmarshal(expireCheck.(json.RawMessage), &skipExpiryCheck)
+							if err != nil {
+								api.EchoInstance().Logger.Errorf("unable to unmarshal skip expiry '%s'", err)
+							}
+						}
+						//create key set and verifier
+						keySet := oidc.NewRemoteKeySet(context.Background(), jwksUrl)
+						tokenVerifier := oidc.NewVerifier(openIdConnectUrl, keySet, &oidc.Config{
+							ClientID:             "",
+							SupportedSigningAlgs: algs,
+							SkipClientIDCheck:    true,
+							SkipExpiryCheck:      skipExpiryCheck,
+							SkipIssuerCheck:      true,
+							Now:                  time.Now,
+						})
+						verifiers = append(verifiers, tokenVerifier)
+					}
+				}
+
+			}
+		}
+
+	}
+
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(ctxt echo.Context) error {
+			var err error
+			var token string
+			if !securityCheck {
+				return next(ctxt)
+			}
+			if len(verifiers) == 0 {
+				api.e.Logger.Errorf("unexpected error no verifiers were set")
+				return NewControllerError("unexpected error no verifiers were set", nil, http.StatusBadRequest)
+			}
+			newContext := ctxt.Request().Context()
+			//get the token from request header since this runs before the context middleware
+			if ctxt.Request().Header[weoscontext.AUTHORIZATION] != nil {
+				token = ctxt.Request().Header[weoscontext.AUTHORIZATION][0]
+			}
+			if token == "" {
+				api.e.Logger.Errorf("no JWT token was found")
+				return NewControllerError("no JWT token was found", nil, http.StatusUnauthorized)
+			}
+			jwtToken := strings.Replace(token, "Bearer ", "", -1)
+			var idToken *oidc.IDToken
+			for _, tokenVerifier := range verifiers {
+				idToken, err = tokenVerifier.Verify(newContext, jwtToken)
+				if err != nil || idToken == nil {
+					api.e.Logger.Errorf(err.Error())
+					return NewControllerError("unexpected error verifying token", err, http.StatusUnauthorized)
+				}
+			}
+
+			newContext = context.WithValue(newContext, weoscontext.USER_ID, idToken.Subject)
+			request := ctxt.Request().WithContext(newContext)
+			ctxt.SetRequest(request)
+			return next(ctxt)
+
+		}
+	}
+}
+
+//GetJwkUrl fetches the jwk url from the open id connect url
+func GetJwkUrl(openIdUrl string) (string, error) {
```

Comment:
This could be a utility function and could probably be used as a part of the check of if the open id URL is valid (I mentioned that the name of the URL shouldn't be used in another comment)